vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -208,22 +198,28 @@ export class WebRequestWorker
 
             return response.data;
         }
-        catch (error)
+        catch (error : any)
         {
             if (throwOnError)
             {
-                let formattedError = error as Error;
-                if ((formattedError.message as string).toLowerCase().includes('block')) {
-                    formattedError = new Error(`Software restriction policy is blocking .NET installation: Request to ${this.url} Failed: ${formattedError.message}`);
+                if(isAxiosError(error))
+                {
+                    const axiosBasedError = error as AxiosError;
+                    const summarizedError = new Error(
+`Request to ${this.url} Failed: ${axiosBasedError.message}. Aborting.
+${axiosBasedError.cause? `Error Cause: ${axiosBasedError.cause!.message}` : ``}
+Please ensure that you are online.
+
+If you're on a proxy and disable registry access, you must set the proxy in our extension settings. See https://github.com/dotnet/vscode-dotnet-runtime/blob/main/Documentation/troubleshooting-runtime.md.`);
+                    this.eventStream.post(new WebRequestError(summarizedError));
+                    throw summarizedError;
                 }
                 else
                 {
-                    formattedError = new Error(`Please ensure that you are online: Request to ${this.url} Failed: ${formattedError.message}.
-If you are on a corporate proxy with a result of 400, you may need to manually set the proxy in our extension settings. Please see https://github.com/dotnet/vscode-dotnet-runtime/blob/main/Documentation/troubleshooting-runtime.md for more details.
-If your proxy requires credentials and the result is 407: we cannot currently handle your request, and you should configure dotnet manually following the above link.`);
+                    const genericError = new Error(`Web Request to ${this.url} Failed: ${error.message}. Aborting. Please ensure that you are online.`);
```

Comment:
```suggestion
                    const genericError = new Error(`Web Request to ${this.url} Failed: ${error.message}. Aborting. Please ensure that you have network connection.`);
```
I think it's great explanations! I wonder if "network connection" is more intuitive, but you have more understanding of the context🙂

Reply:
Thank you, I think this is a good suggestion, but I always try to keep errors as small as possible, as the longer the message is, the less likely someone is to read them.